components/service/glean/scripts/sdk_generator.gradle

```diff
@@ -44,17 +71,39 @@ subprocess.check_call([
 
 // Configure the Python environments
 envs {
-    bootstrapDirectory = new File(buildDir, 'bootstrap')
-    envsDirectory = new File(buildDir, 'envs')
+    bootstrapDirectory = GLEAN_BOOTSTRAP_DIR
     pipInstallOptions = "--trusted-host pypi.python.org --no-cache-dir"
 
     // Setup a miniconda environment. conda is used because it works
     // non-interactively on Windows, unlike the other options
-    conda "Miniconda3", "Miniconda3-4.5.11", "64", ["glean_parser==${GLEAN_PARSER_VERSION}"]
+    conda "Miniconda3", "Miniconda3-${MINICONDA_VERSION}", "64", ["glean_parser==${GLEAN_PARSER_VERSION}"]
 }
 
+// Even though we are installing the Miniconda environment to the gradle user
+// home directory, the gradle-python-envs plugin is hardcoded to download the
+// installer to the project's build directory. Doing so will fail if the
+// project's build directory doesn't already exist. This task ensures that
+// the project's build directory exists before downloading and installing the
+// Miniconda environment.
+// See https://github.com/JetBrains/gradle-python-envs/issues/26
+task createBuildDir {
+    description = "Make sure the build dir exists before creating the Python Environments"
+    onlyIf {
+        !file(buildDir).exists()
+    }
+    doLast {
+        println "Creating build directory:" + buildDir.getPath()
+        buildDir.mkdir()
+    }
+}
+
+preBuild.dependsOn(createBuildDir)
 preBuild.finalizedBy("build_envs")
 
+clean {
+    delete GLEAN_BOOTSTRAP_DIR
+}
```

Comment:
We execute `clean` for a bunch of reasons. I wonder if we should give that a different name?

Also: Doesn't that need to be wrapped in `doLast` to not get executed whenever the script gets evaluated (Not sure about that)?